internal/vm/vm.go

```diff
@@ -29,8 +30,19 @@ func init() {
 		LuaLDir + "/?/init.lua"
 }
 
-func New() (*VM, error) {
-	l := lua.NewState(lua.Options{FS: &fs.FS{EmbedFS: goks.LuaTree}})
+func New(injectFS *embed.FS) (*VM, error) {
```

Comment:
Please move this into a VMOpts struct.
We may have additions in the future.
Please also add a one-line comment on the purpose of the injectable FS.

Reply:
**Note**: linter doesn't like `vm.VMOpts` so I kept it as `Opts`.